src/text_parser.py

```diff
@@ -0,0 +1,203 @@
+    pattern = r'''(`.+?`)'''
+    result = re.sub(pattern, "InlineCode", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def remove_empty_space(text):
+    pass
+
+def sub_PlusOne(text):
+    pattern = r"""(\+1)"""
+    result = re.sub(pattern,"plus one", text,0,re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+#remove @
+def replace_mention(text):
+    pattern = r'''@[A-Za-z0-9-]+''' # remove @
+    result = re.sub(pattern, "@friend", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+
+def remove_url(text):
+    pattern = r'''https?://[^ ]+'''
+    result = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def remove_newline(text):
+    #text = text.encode('utf-8').replace('\n', ' ').replace('\r', '').replace('\t','')
+    text = text.replace('\n', ' ').replace('\r', '').replace('\t', '')
+    return text
+
+# just remove all :
+def remove_emoji_marker(text):
+
+    pattern = r''':'''
+    text = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return text
+
+#has probelms. Deal with this later. Removing letters that are not supposed to be removed
+def remove_emoji(text):
+    pattern = r"""
+        [\U0001F600-\U0001F64F]|[\U0001F300-\U0001F5FF]|[\U0001F680-\U0001F6FF]|[\U0001F1E0-\U0001F1FF]|[\U00002702-\U000027B0]|[\U000024C2-\U0001F251]
+        """
+    result = re.sub(pattern,"",text,0,re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+#################### Extracting textual features ####################
+
+# get length of text in tokens
+def get_length(text):
+    return len(text)
+
+
+# get average length of word
+def get_avg_length(text):
+    word_length_lst = [len(word) for word in text.split(' ')]
+    return sum(word_length_lst) / len(word_length_lst)
+
+
+# get number of punctuations
+def count_punct(text):
+    pattern = r'[!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~]'
+    sentences = re.findall(pattern, text)
+    return len(sentences)
+
+
+# get number of periods, question marks,quotes, and repeated punctuation
+def count_QEMark(text):
+    pattern = r'[!?]'
+    return len(re.findall(pattern, text))
+
+
+# get number of 1 letter tokens
+def count_one_letter(text):
+    # split by space
+    pattern = r'(^|\W)[a-zA-Z](\W|\Z)'
+    l = re.findall(pattern, text)
+    return len(l)
+
+
+# get number of capitalized letters
+def count_captial(text):
+    pattern = r'[A-Z]'
+    return len(re.findall(pattern, text))
+
+
+# get number of URLs
+def count_url(text):  # check for http:// or https://
+    pattern = r'https?://[^ ]+'
+    return len(re.findall(pattern, text))
+
+
+# get number of tokens with non-alpha characters in the middle???
+def count_non_alpha_in_middle(text):
+    pattern = r'(^|\w)[!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~](\w|\Z)'
+    return len(re.findall(pattern, text))
+
+# get number of politeness words
+def count_politeness_words(text):
+    pass
```

Comment:
rm, false sense of security

Reply:
again, not used.. deleted. we will use convokit for counting politeness words anyway..